kolibri/plugins/app/utils.py

```diff
@@ -16,25 +22,32 @@ def __init__(self):
         self._capabilities = {}
 
     def __contains__(self, capability):
-        return capability in self._capabilities
+        return self.enabled and (capability in self._capabilities)
 
     def register(self, **kwargs):
         for capability in CAPABILITES:
             if capability in kwargs:
                 self._capabilities[capability] = kwargs[capability]
 
-    def get_initialize_url(self, next_url=None):
+    def get_initialize_url(self, next_url=None, auth_token=None):
         if not self.enabled:
             raise RuntimeError("App plugin is not enabled")
         # Import here to prevent a circular import
         from kolibri.core.device.models import DeviceAppKey
 
         url = reverse(
-            "kolibri:kolibri.plugins.app:initialize", args=(DeviceAppKey.get_app_key(),)
+            "kolibri:kolibri.plugins.app:initialize",
+            args=(DeviceAppKey.get_app_key(),),
```

Comment:
So... is it always still using this centralized token, vs some user-specific token that the app frontend app itself gets? I'm thinking about the EOS case of the backend service running as a different user than the app frontend (of which there may be multiple, under different users).

Reply:
Oh, wait, so there are actually two tokens? The app key and the auth token? Where are the auth tokens generated/stored, and how does the app logic validate them?

Reply:
The app wrapper is responsible for ensuring the auth token is generated/stored. In the case of the Android app, I think we can just handle this on disk in a JSON file or somesuch.

For the Endless app - there will be the separate system service that @dylanmccall suggested for dishing out the tokens to the currently active user.

This seemed to be what was needed in order to be able to distinguish the currently running app user vs the user for the server (which might not be the app user in the Endless case).